http-server/src/server.rs

```diff
@@ -178,56 +218,26 @@ impl<M> Builder<M> {
 	///       occupied_addr,
 	///       "127.0.0.1:0".parse().unwrap(),
 	///   ];
-	///   assert!(jsonrpsee_http_server::HttpServerBuilder::default().build(occupied_addr).is_err());
-	///   assert!(jsonrpsee_http_server::HttpServerBuilder::default().build(addrs).is_ok());
+	///   assert!(jsonrpsee_http_server::HttpServerBuilder::default().build(occupied_addr).await.is_err());
+	///   assert!(jsonrpsee_http_server::HttpServerBuilder::default().build(addrs).await.is_ok());
 	/// }
 	/// ```
-	pub fn build(self, addrs: impl ToSocketAddrs) -> Result<Server<M>, Error> {
-		let mut err: Option<Error> = None;
-
-		for addr in addrs.to_socket_addrs()? {
-			let (listener, local_addr) = match self.inner_builder(addr) {
-				Ok(res) => res,
-				Err(e) => {
-					err = Some(e);
-					continue;
-				}
-			};
+	pub async fn build(self, addrs: impl ToSocketAddrs) -> Result<Server<M>, Error> {
+		let listener = TcpListener::bind(addrs).await?.into_std()?;
 
-			return Ok(Server {
-				listener,
-				local_addr,
-				access_control: self.access_control,
-				max_request_body_size: self.max_request_body_size,
-				max_response_body_size: self.max_response_body_size,
-				resources: self.resources,
-				tokio_runtime: self.tokio_runtime,
-				middleware: self.middleware,
-			});
-		}
-
-		let err = err.unwrap_or_else(|| std::io::Error::new(std::io::ErrorKind::NotFound, "No address found").into());
-		Err(err)
-	}
-
-	fn inner_builder(
-		&self,
-		addr: SocketAddr,
-	) -> Result<(hyper::server::Builder<hyper::server::conn::AddrIncoming>, Option<SocketAddr>), Error> {
-		let domain = Domain::for_address(addr);
-		let socket = Socket::new(domain, Type::STREAM, None)?;
-		socket.set_nodelay(true)?;
-		socket.set_reuse_address(true)?;
-		socket.set_nonblocking(true)?;
-		socket.set_keepalive(self.keep_alive)?;
-		let address = addr.into();
-		socket.bind(&address)?;
-
-		socket.listen(1024)?;
-		let listener: TcpListener = socket.into();
 		let local_addr = listener.local_addr().ok();
-		let listener = hyper::Server::from_tcp(listener)?;
-		Ok((listener, local_addr))
+		let listener = hyper::Server::from_tcp(listener)?.tcp_nodelay(true);
```

Comment:
in the comment above, you say that hyper sets tcp_nodelay == true, so is this `nodelay(true)` going to be ignored anyway?

Reply:
yeah, I was thinking that we might want to expose these settings in the builder.

It's like a warning that even if folk configure these on their own sockets it might get overwritten but let's revert this then.

Reply:
nah, my comment was wrong hyper sets it to false